### PR TITLE
SAAS-20313/Front-Door-WAF-Logs-Enabled

### PR DESCRIPTION
**Pull request by alphadev4**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -1478,4 +1479,4 @@ module.exports = {
         'securityNotificationsEnabled'  : require(__dirname + '/plugins/alibaba/securitycenter/securityNotificationsEnabled.js'),
         'vulnerabilityScanEnabled'      : require(__dirname + '/plugins/alibaba/securitycenter/vulnerabilityScanEnabled.js')
     }
-};
+};
```
Comment (alphadev4) [inline exports.js:1482]:
```suggestion
};

```

File: plugins/azure/applicationGateway/agAccessLogsEnabled.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Application Gateway Access Logs Enabled',
+    category: 'Application Gateway',
+    domain: 'Network Access Control',
+    description: 'Ensures that Application Gateway Access Log is enabled.',
```
Comment (mehakseedat63) [inline plugins/azure/applicationGateway/agAccessLogsEnabled.js:8]:
```suggestion
    description: 'Ensures that Application Gateway access logs are enabled.',
```

File: plugins/azure/applicationGateway/agAccessLogsEnabled.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    more_info: 'Application Gateway logs provide detailed information for events related to a resource and its operations. Access logs helps to analyze important information including the caller\'s IP, requested URL, response latency, return code, and bytes in and out. An access log is collected every 60 seconds.',
+    recommended_action: 'Ensure that diagnostic setting for Application Gateway Access Log is enabled.',
```
Comment (mehakseedat63) [inline plugins/azure/applicationGateway/agAccessLogsEnabled.js:10]:
```suggestion
    recommended_action: 'Ensure that diagnostic setting for Application Gateway access logs is enabled.',
```

File: plugins/azure/applicationGateway/agAccessLogsEnabled.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+            applicationGateways.data.forEach(function(appGateway) {
+                if (!appGateway.id) return;
+                const diagnosticSettings = helpers.addSource(cache, source,
+                    ['diagnosticSettings', 'listByApplicationGateways', location, appGateway.id]);
+
+                if (!diagnosticSettings || diagnosticSettings.err || !diagnosticSettings.data) {
+                    helpers.addResult(results, 3, 'Unable to query Application Gateway diagnostics settings: ' + helpers.addError(diagnosticSettings), location, appGateway.id);
+                } else if (!diagnosticSettings.data.length) {
+                    helpers.addResult(results, 2, 'No existing Application Gateway diagnostics settings found', location, appGateway.id);
```
Comment (mehakseedat63) [inline plugins/azure/applicationGateway/agAccessLogsEnabled.js:43]:
```suggestion
                } else if (!diagnosticSettings.data.length) {
                    helpers.addResult(results, 2, 'No existing Application Gateway diagnostics settings found', location, appGateway.id);
```
I think not found means that it's not enabled so we don't have to add this

File: plugins/azure/applicationGateway/agFirewallLogsEnabled.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Application Gateway Firewall Logs Enabled',
+    category: 'Application Gateway',
+    domain: 'Network Access Control',
+    description: 'Ensures that Application Gateway Firewall Log is enabled.',
```
Comment (mehakseedat63) [inline plugins/azure/applicationGateway/agFirewallLogsEnabled.js:8]:
```suggestion
    description: 'Ensures that Application Gateway Firewall logs are enabled.',
```

File: plugins/azure/applicationGateway/agFirewallLogsEnabled.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    more_info: 'Application Gateway logs provide detailed information for events related to a resource and its operations. Firewall logs helps to analyze the requests that are logged through either detection or prevention mode of an application gateway that is configured with the web application firewall.',
+    recommended_action: 'Ensure that diagnostic setting for Application Gateway Firewall Log is enabled.',
```
Comment (mehakseedat63) [inline plugins/azure/applicationGateway/agFirewallLogsEnabled.js:10]:
```suggestion
    recommended_action: 'Ensure that diagnostic setting for Application Gateway Firewall logs is enabled.',
```

File: plugins/azure/frontdoor/wafLogsEnabled.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Front Door WAF Logs Enabled',
+    category: 'Front Door',
+    domain: 'Content Delivery',
+    description: 'Ensures that Azure Front Door WAF Log is enabled.',
```
Comment (mehakseedat63) [inline plugins/azure/frontdoor/wafLogsEnabled.js:8]:
```suggestion
    description: 'Ensures that Azure Front Door WAF logs are enabled.',
```

File: plugins/azure/frontdoor/wafLogsEnabled.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    more_info: 'Azure Front Door captures several types of logs. Web application firewall (WAF) logs can be used to detect potential attacks, and false positive detections that might indicate legitimate requests that the WAF blocked.',
+    recommended_action: 'Ensure that diagnostic setting for Front Door WAF Log is enabled.',
```
Comment (mehakseedat63) [inline plugins/azure/frontdoor/wafLogsEnabled.js:10]:
```suggestion
    recommended_action: 'Ensure that diagnostic setting for Front Door WAF logs is enabled.',
```

File: plugins/azure/frontdoor/wafLogsEnabled.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    link: 'https://learn.microsoft.com/en-us/azure/frontdoor/standard-premium/how-to-logs',
```
Comment (mehakseedat63) [inline plugins/azure/frontdoor/wafLogsEnabled.js:11]:
```suggestion
    link: 'https://learn.microsoft.com/en-us/azure/web-application-firewall/afds/waf-front-door-monitor?pivots=front-door-standard-premium',
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: helpers/azure/functions.js [hunk 10]
```diff
@@ -702,5 +702,5 @@ module.exports = {
     remediatePlugin: remediatePlugin,
     processCall: processCall,
     remediateOpenPorts: remediateOpenPorts,
-    remediateOpenPortsHelper: remediateOpenPortsHelper
+    remediateOpenPortsHelper: remediateOpenPortsHelper,
```
Comment (alphadev4) [inline helpers/azure/functions.js:705]:
```suggestion
    remediateOpenPortsHelper: remediateOpenPortsHelper
```

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**